pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -480,7 +482,7 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 			hivemetrics.GetClusterDeploymentType(cd)).Set(
 			time.Since(cd.CreationTimestamp.Time).Seconds())
 
-		job, err := install.GenerateInstallerJob(
+		job, pvc, err := install.GenerateInstallerJob(
```

Comment:
I would prefer to see the PVC created separately from the installer job and the PVC passed in as a parameter to `GenerateInstallerJob`. The creation of the PVC is really a separate action from creating the job. It has a direct relationship with the clusterdeployment and only a passing relationship with the job--in much the same way that the serviceaccount is creating separately from the job.